nemo/src/model/types/primitive_logical_value.rs

```diff
@@ -341,33 +343,126 @@ impl TryFrom<Term> for PhysicalString {
             Term::RdfLiteral(RdfLiteral::DatatypeValue {
                 ref value,
                 ref datatype,
-            }) => {
-                match datatype.as_ref() {
-                    XSD_STRING => Ok(LogicalString(value.to_string()).into()),
-                    XSD_INTEGER => Ok(LogicalInteger(value.parse().map_err(|_err| {
-                        InvalidRuleTermConversion::new(term, PrimitiveType::Any)
-                    })?)
-                    .into()),
-                    XSD_DECIMAL => {
-                        let (a, b) = value
-                            .rsplit_once('.')
-                            .and_then(|(a, b)| Some((a.parse().ok()?, b.parse().ok()?)))
-                            .or_else(|| Some((value.parse().ok()?, 0)))
-                            .ok_or(InvalidRuleTermConversion::new(term, PrimitiveType::Any))?;
-
-                        Ok(Decimal(a, b).into())
+            }) => match datatype.as_ref() {
+                XSD_STRING => Ok(LogicalString(value.to_string()).into()),
+                XSD_INTEGER => match integer(span_from_str(value)) {
+                    Ok((remainder, number)) => {
+                        if !remainder.is_empty() {
+                            Err(InvalidRuleTermConversion::new(term, PrimitiveType::Any))
+                        } else {
+                            match number {
+                                NumericLiteral::Integer(i) => Ok(LogicalInteger::from(i).into()),
+                                _ => unreachable!(
+                                    "We only parse as integer so only the above case can occur."
+                                ),
+                            }
+                        }
+                    }
+                    Err(nom::Err::Error(located_parse_error)) => {
+                        let parse_error: &ParseError = located_parse_error.get_last_inner().into();
+
+                        match parse_error {
+                            ParseError::SyntaxError(_) => {
+                                Err(InvalidRuleTermConversion::new(term, PrimitiveType::Any))
+                            }
+                            _ => Ok(DatatypeValue(value.to_string(), datatype.to_string()).into()),
+                        }
+                    }
+                    _ => Err(InvalidRuleTermConversion::new(term, PrimitiveType::Any)),
+                },
+                XSD_DECIMAL => match nom::branch::alt((decimal, integer))(span_from_str(value)) {
+                    Ok((remainder, number)) => {
+                        eprintln!("{remainder:?};{number:?}");
```

Comment:
I guess those should still go (or, at the very least, become `log::debug!` invocations) ;)

Reply:
Just forgot to remove those, thx :)